### PR TITLE
Add REST filter chain

### PR DESCRIPTION
**Pull request by meriouma**
Thinking about it, maybe we could simply remove the Interceptors. With the filter you can either call `chain.doFilter()` to continue the chain, or `command.execute()` to stop and return

### CODE REVIEW

**Comment by olafleur** [comment]
LGTM

File: gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/client/filter/DefaultRestFilterChain.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    public DispatchRequest doFilter(
+            RestAction<?> action,
+            RestCallback<?> resultCallback,
+            ExecuteCommand<RestAction<?>, RestCallback<?>> executeCommand) {
+
+        if (filterIterator.hasNext()) {
+            Map.Entry<RestContext, RestFilter> filterEntry = filterIterator.next();
+            RestContext subjectContext = new RestContext.Builder((RestAction) action).build();
+
+            if (filterEntry.getKey().equals(subjectContext)) {
```
Comment (Chris-V) [inline gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/client/filter/DefaultRestFilterChain.java]: would help to extract this in a method and pass key and value as separate parameters

File: gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/client/gin/BaseRestDispatchModuleBuilder.java [hunk 2]
```diff
@@ -145,6 +153,30 @@ public B interceptorRegistry(Class<? extends RestInterceptorRegistry> intercepto
...
+    }
+
+    /**
+     * Specify an alternate REST filter chain.
+     *
+     * @param filterRegistry A {@link com.gwtplatform.dispatch.rest.client.filter.RestFilterChain} class.
+     *
+     * @return this {@link com.gwtplatform.dispatch.rest.client.gin.BaseRestDispatchModuleBuilder builder} object.
+     */
+    public B flterChain(Class<? extends RestFilterChain> filterChain) {
```
Comment (Chris-V) [inline gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/client/gin/BaseRestDispatchModuleBuilder.java]: I think I would move this one to the CoreModule / Builder

**Comment by Chris-V** [comment]
Agree, filters do the same job (and more) as the interceptors here. We should remove them.

**Comment by Chris-V** [comment]
LGTM

**Pull request merged**